pkg/ingester/ingester.go

```diff
@@ -2989,3 +3046,23 @@ func (i *Ingester) UserRegistryHandler(w http.ResponseWriter, r *http.Request) {
 		Timeout:            10 * time.Second,
 	}).ServeHTTP(w, r)
 }
+
+// checkReadOverloaded checks whether the ingester read path is overloaded wrt. CPU and/or memory.
+func (i *Ingester) checkReadOverloaded() error {
+	if !i.cfg.utilizationLimitsEnabled {
```

Comment:
[nit] Since I suggested to remove `cfg.utilizationLimitsEnabled` you can change this as:

```suggestion
	if i. utilizationBasedLimiter != nil {
```